### PR TITLE
update: Added sed substitution of the last message

### PR DESCRIPTION
**Pull request by mrshu**
* Added s/pattern/repl/ type of substitutions for in-channel commands
  and messages directed at a bot.

Signed-off-by: mr.Shu <mr@shu.io>

### CODE REVIEW

File: basic.py [hunk 1]
```diff
@@ -90,3 +91,33 @@ def eval_(event):
...
+last_events = {}
+
+
+@event
+def sub_event_catcher(event):
+    if event.event_type in ['message', 'cmd']:
+        last_events[event.meta['host']] = event
+
+
+@match(regex=r'^s/(.*?)/(.*?)/([a-z]*)$')
```
Comment (pepol) [inline basic.py]: @mrshu what if we want to use `/` inside the matched string? Wouldn't it be better to do it just like `sed` does?

File: basic.py [hunk 2]
```diff
@@ -90,3 +91,33 @@ def eval_(event):
...
+def sub_match(event, pattern, repl, flags, *args, **kwargs):
```
Comment (pepol) [inline basic.py]: `repl` reminds me more of `REPL` than `replacement`. I wonder if there isn't a better name for that argument.

**Comment by pepol** [comment]
@mrshu generally looks good, but I wonder if there is something we can do about the slashes.

File: basic.py (hunk 1 above)
Comment (mrshu) [inline basic.py]: @pepol like what? Any two separators? That doesn't seem to be matchable by a regular expression. We could in theory match just that initial `s` and try to parse anything afterwards but that kind of breaks the purpose of `@match`.

File: basic.py (hunk 2 above)
Comment (mrshu) [inline basic.py]: In this situation I do not think it matters. If you do find a better name, do let me know.

File: basic.py (hunk 1 above)
Comment (pepol) [inline basic.py]: Well I'd suggest to add `|` as well, at least... or something similar...

File: basic.py (hunk 2 above)
Comment (pepol) [inline basic.py]: Well it confused me when I first took a look at the code (actually at example) for like 0.8s so... :-D I'd suggest changing it. But it's your call (I have only thought of spelling it out in full: `replacement`).

**Comment by mrshu** [comment]
@pepol So I did update those things in the last two commit, along with a simple implementation of `g` and `i` flags (it doesn't make too much sense to add others).

cc @Adman @JakubNvk

**Comment by pepol** [comment]
lgtm! Thanks @mrshu! Please squash.

**Comment by mrshu** [comment]
@pepol should be done by now.

**Pull request merged**